### DISCUSSION TITLE
Sveltekit Dependency Issue with Vercel

### DISCUSSION THREAD

**Discussion by gabeatwell** [vercel/community#6677] [Help]
### Summary

I have deployed a sveltekit project to vercel and the website is deployed, but lately when I redeploy it, I get an npm error.. It says it has to do with the sveltekit versions so i tried to update to 2.5.7 and downgrade to 1.0.0 and nothing worked. I also tried removing my node_modules folder and npm installing it again and it didnt work.

### Example

https://www.github.com/gabrielatwell1987/portfolio

### Steps to Reproduce

npm ERR! code ERESOLVE
--
13:14:53.804 | npm ERR! ERESOLVE could not resolve
13:14:53.805 | npm ERR!
13:14:53.805 | npm ERR! While resolving: @sveltejs/adapter-auto@2.1.1
13:14:53.805 | npm ERR! Found: @sveltejs/kit@2.5.7
13:14:53.806 | npm ERR! node_modules/@sveltejs/kit
13:14:53.806 | npm ERR!   dev @sveltejs/kit@"^2.5.7" from the root project
13:14:53.806 | npm ERR!   peer @sveltejs/kit@"^2.4.0" from @sveltejs/adapter-vercel@5.3.0
13:14:53.806 | npm ERR!   node_modules/@sveltejs/adapter-vercel
13:14:53.806 | npm ERR!     dev @sveltejs/adapter-vercel@"^5.3.0" from the root project
13:14:53.807 | npm ERR!
13:14:53.807 | npm ERR! Could not resolve dependency:
13:14:53.807 | npm ERR! peer @sveltejs/kit@"^1.0.0" from @sveltejs/adapter-auto@2.1.1
13:14:53.807 | npm ERR! node_modules/@sveltejs/adapter-auto
13:14:53.808 | npm ERR!   dev @sveltejs/adapter-auto@"^2.0.0" from the root project
13:14:53.808 | npm ERR!
13:14:53.808 | npm ERR! Conflicting peer dependency: @sveltejs/kit@1.30.4
13:14:53.808 | npm ERR! node_modules/@sveltejs/kit
13:14:53.808 | npm ERR!   peer @sveltejs/kit@"^1.0.0" from @sveltejs/adapter-auto@2.1.1
13:14:53.808 | npm ERR!   node_modules/@sveltejs/adapter-auto
13:14:53.809 | npm ERR!     dev @sveltejs/adapter-auto@"^2.0.0" from the root project
13:14:53.809 | npm ERR!
13:14:53.809 | npm ERR! Fix the upstream dependency conflict, or retry
13:14:53.809 | npm ERR! this command with --force or --legacy-peer-deps
13:14:53.809 | npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
13:14:53.809 | npm ERR!
13:14:53.809 | npm ERR!
13:14:53.809 | npm ERR! For a full report see:
13:14:53.809 | npm ERR! /vercel/.npm/_logs/2024-04-29T20_14_52_398Z-eresolve-report.txt
13:14:53.810 |  
13:14:53.810 | npm ERR! A complete log of this run can be found in: /vercel/.npm/_logs/2024-04-29T20_14_52_398Z-debug-0.log
13:14:53.821 | Error: Command "npm install" exited with 1

### System Info

```shell
HP Pavilion Gaming Desktop TG01-2xxx, Windows 11, Brave Browser
```

### Severity

serious

**Comment by amyegan** [comment]
Hey @gabrielatwell1987. It looks like you have some conflicting dependencies to work out. I get the same error if I try to `npm install` in a local copy of that repo

**Reply by gabeatwell** [reply]
I don't know how to work them out though.. What do you suggest?

**Reply by gabeatwell** [reply]
I'm not sure as to why my vite version isn't compatible with my sveltekit version..

**Reply by amyegan** [reply]
I was able get it to deploy by changing the `package.json` to
```json
{
	"name": "portfolio",
	"version": "0.0.5",
	"private": true,
	"scripts": {
		"dev": "vite dev",
		"build": "vite build",
		"preview": "vite preview",
		"lint": "prettier --plugin-search-dir . --check . && eslint .",
		"format": "prettier --plugin-search-dir . --write .",
		"test": "vitest"
	},
	"devDependencies": {
		"@sveltejs/adapter-vercel": "^5.3.0",
		"@sveltejs/kit": "^2.5.7",
		"eslint": "^8.57.0",
		"eslint-config-prettier": "^8.10.0",
		"eslint-plugin-svelte": "^2.38.0",
		"mdsvex": "^0.11.0",
		"prettier": "^2.8.8",
		"prettier-plugin-svelte": "^2.10.1",
		"svelte-carousel": "^1.0.25",
		"svelte-debug-console": "^0.0.2",
		"svelte-hamburgers": "^4.2.0",
		"vite": "^5.2.10",
		"vitest": "^1.5.2"
	},
	"type": "module",
	"dependencies": {
		"@analytics/google-analytics": "^1.0.7",
		"@vercel/analytics": "^1.2.2",
		"dat.gui": "^0.7.9",
		"from": "^0.1.7",
		"gsap": "^3.12.5",
		"highlight.js": "^11.9.0",
		"lucide-svelte": "^0.284.0",
		"split-type": "^0.3.4",
		"three": "^0.161.0"
	}
}

```

**Reply by amyegan** [reply]
Went back to the last version of package.json that was known to work and double checked that everything would install as expected. Then upgraded vite to latest. Then I removed the svelte packages before updating them to avoid conflicts

**Reply by gabeatwell** [reply]
Thank you for your help. I did what you said and also reinstalled node and it works now.
